src/base/UniswapV3.sol

```diff
@@ -0,0 +1,390 @@
+
+        uint256 amount0;
+        uint256 amount1;
+        (position.id, position.liquidity, amount0, amount1) = POSITION_MANAGER.mint(
+            MintParams({
+                token0: position.tokens[0],
+                token1: position.tokens[1],
+                fee: position.fee,
+                tickLower: position.tickLower,
+                tickUpper: position.tickUpper,
+                amount0Desired: amount0Desired,
+                amount1Desired: amount1Desired,
+                amount0Min: 0,
+                amount1Min: 0,
+                recipient: address(this),
+                deadline: block.timestamp
+            })
+        );
+
+        balances[0] -= amount0;
+        balances[1] -= amount1;
+    }
+
+    /* ///////////////////////////////////////////////////////////////
+                    INCREASE LIQUIDITY LOGIC
+    /////////////////////////////////////////////////////////////// */
+
+    /**
+     * @notice Swaps one token for another to rebalance the Liquidity Position.
+     * @param balances The balances of the underlying tokens.
+     * param positionManager The contract address of the Position Manager.
+     * @param position A struct with position and pool related variables.
+     * @param amount0Desired The desired amount of token0 to add as liquidity.
+     * @param amount1Desired The desired amount of token1 to add as liquidity.
+     */
+    function _increaseLiquidity(
+        uint256[] memory balances,
+        address,
+        PositionState memory position,
+        uint256 amount0Desired,
+        uint256 amount1Desired
+    ) internal virtual override {
+        ERC20(position.tokens[0]).safeApproveWithRetry(address(POSITION_MANAGER), amount0Desired);
+        ERC20(position.tokens[1]).safeApproveWithRetry(address(POSITION_MANAGER), amount1Desired);
+
+        uint256 amount0;
+        uint256 amount1;
+        (position.liquidity, amount0, amount1) = POSITION_MANAGER.increaseLiquidity(
+            IncreaseLiquidityParams({
+                tokenId: position.id,
+                amount0Desired: amount0Desired,
+                amount1Desired: amount1Desired,
+                amount0Min: 0,
+                amount1Min: 0,
+                deadline: block.timestamp
+            })
+        );
+
+        balances[0] -= amount0;
+        balances[1] -= amount1;
+    }
+
+    /* ///////////////////////////////////////////////////////////////
+                          STAKING LOGIC
+    /////////////////////////////////////////////////////////////// */
+
+    /**
+     * @notice Stakes a Liquidity Position.
+     * param balances The balances of the underlying tokens.
+     * param positionManager The contract address of the Position Manager.
+     * param position A struct with position and pool related variables.
+     */
+    function _stake(uint256[] memory, address, PositionState memory) internal virtual override { }
+}
```

Comment:
what's the diff with slipstream in terms of native eth? Uniswap base doesn't have a receive function, slipstream base does

Reply:
Slipstream sends native eth back if it is on the position manager on every increase and decrease liquidity.
Someone could use it to grieve attack.
![image](https://github.com/user-attachments/assets/da677a37-7c38-4557-b4e0-86f65d3ad168)